controllers/reconcilers/logging_installation/logging_installation_reconciler.go

```diff
@@ -0,0 +1,283 @@
+package logging_installation
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	v14 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	v1 "github.com/redhat-developer/observability-operator/v3/api/v1"
+	"github.com/redhat-developer/observability-operator/v3/controllers/model"
+	"github.com/redhat-developer/observability-operator/v3/controllers/reconcilers"
+	v12 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Reconciler struct {
+	client client.Client
+	logger logr.Logger
+}
+
+func NewReconciler(client client.Client, logger logr.Logger) reconcilers.ObservabilityReconciler {
+	return &Reconciler{
+		client: client,
+		logger: logger,
+	}
+}
+func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.Observability) (v1.ObservabilityStageStatus, error) {
+	if cr.DescopedModeEnabled() {
+		return v1.ResultSuccess, nil
+	}
+
+	subscription := model.GetLoggingSubscription(cr)
+	err := r.client.Delete(ctx, subscription)
+	if err != nil && !errors.IsNotFound(err) {
+		return v1.ResultFailed, err
+	}
+
+	// Delete csv to uninstall
+	csv := &v1alpha1.ClusterServiceVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster-logging.5.5.3",
+			Namespace: "openshift-logging",
+		},
+	}
+	err = r.client.Delete(ctx, csv)
+	if err != nil && !errors.IsNotFound(err) {
+		return v1.ResultFailed, err
+	}
+
+	return v1.ResultSuccess, nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, cr *v1.Observability, s *v1.ObservabilityStatus) (v1.ObservabilityStageStatus, error) {
+	if cr.DescopedModeEnabled() {
+		return v1.ResultSuccess, nil
+	}
+
+	// First let's check if there is already a cluster-logging-operator
+	// If the operator is already installed we want to leave it in place for now
+	deployments := &v12.DeploymentList{}
+	opts := &client.ListOptions{
+		Namespace: "openshift-logging",
+	}
+	err := r.client.List(ctx, deployments, opts)
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+	var foundLoggingOperator = false
+
+	for _, deployment := range deployments.Items {
+		if strings.HasPrefix(deployment.Name, "cluster-logging-operator") {
+			foundLoggingOperator = true
+			r.logger.Info("Found cluster-logging-operator: skipping install")
```

Comment:
Would be better to not log that cluster-logging-operator is found every reconcile. Can just be:
`return v1.ResultSuccess, nil`

Reply:
Yep good point, I've added that check.